master/internal/api_project.go

```diff
@@ -318,6 +318,267 @@ func (a *apiServer) getProjectColumnsByID(
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "resourcePool",
+			DisplayName: "Resource pool",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "checkpointSize",
+			DisplayName: "Checkpoint size",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "checkpointCount",
+			DisplayName: "Checkpoints",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "user",
+			DisplayName: "User",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "searcherMetric",
+			DisplayName: "Searcher Metric",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "searcherMetricsVal",
+			DisplayName: "Searcher Metric Value",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "externalExperimentId",
+			DisplayName: "External Experiment ID",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "externalRunId",
+			DisplayName: "External Run ID",
+			Location:    projectv1.LocationType_LOCATION_TYPE_RUN,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+	}
+
+	hyperparameters := []struct {
+		WorkspaceID     int
+		Hyperparameters expconf.HyperparametersV0
+		RunID           *int
+	}{}
+
+	// get all runs in project
+	runsQuery := db.Bun().NewSelect().
+		ColumnExpr("?::int as workspace_id", p.WorkspaceId).
+		ColumnExpr("hparams as hyperparameters").
+		ColumnExpr("id as run_id").
+		Table("runs").
+		Where("hparams IS NOT NULL").
+		Where("project_id = ?", id).
+		Order("id")
+
+	runsQuery, err = exputil.AuthZProvider.Get().FilterExperimentsQuery(
+		ctx,
+		curUser,
+		p,
+		db.Bun().NewSelect().TableExpr("(?) AS subq", runsQuery),
+		[]rbacv1.PermissionType{rbacv1.PermissionType_PERMISSION_TYPE_VIEW_EXPERIMENT_METADATA},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	err = runsQuery.Scan(ctx, &hyperparameters)
+	if err != nil {
+		return nil, err
+	}
+
+	runIDs := make([]int, 0, len(hyperparameters))
+	for _, hparam := range hyperparameters {
+		if hparam.RunID != nil {
+			runIDs = append(runIDs, *hparam.RunID)
+		}
+	}
+	summaryMetrics := []struct {
+		MetricName string
+		JSONPath   string
+		MetricType string
+		Count      int32
+	}{}
+
+	if len(runIDs) > 0 {
+		subQuery := db.BunSelectMetricGroupNames().ColumnExpr(
+			`summary_metrics->jsonb_object_keys(summary_metrics)->
+		jsonb_object_keys(summary_metrics->jsonb_object_keys(summary_metrics))->>'type'
+		AS metric_type`).
```

Comment:
can we add a performance test covering this?

Reply:
added one test for each table type.

Reply:
Do you know if we can improve this at all? It looks like an average of 50 seconds.

```
Get Project Columns for Runs Table {'test_name': 'Get Project Columns for Runs Table', 'avg': 51845.51096350322, 'min': 13163.07077, 'med': 60000.411748, 'max': 60001.197938, 'p90': 60000.985919599996, 'p95': 60001.1124485, 'passes': 64, 'fails': 91}
```